charts/mimir-distributed/templates/ingester/ingester-statefulset.yaml

```diff
@@ -16,7 +16,7 @@ spec:
       {{- include "mimir.ingesterSelectorLabels" . | nindent 6 }}
   updateStrategy:
     {{- toYaml .Values.ingester.statefulStrategy | nindent 4 }}
-  serviceName: {{ template "mimir.fullname" . }}-ingester-headless
+  serviceName: {{ template "mimir.fullname" . }}-ingester{{- if not .Values.useGEMLabels -}}-headless{{- end -}}
```

Comment:
it's only headless for OSS? did I read that write, not sure I understand why?

Reply:
upgrade does not work from 1.8.1 GEM otherwise, since serviceName is immutable